juju/juju
# SettingsAddress should take the private-address from the default binding

## Description of change
SettingsAddress must work in the same way as unit-get private-address - that is be space aware and take the address from the space for the default binding, not the machine PrivateAddress

## QA steps
In an environment with spaces, deploy an app "app1" with multiple spaces and default binding set to a space NOT containing the machines' private address. 
Deploy app "app2" and relate it to app1.
Then:
```
juju run --unit app1/0 'unit-get private-address'
should give the same result as 
juju run --unit app2/0 'relation-get -r `relation-ids RELATION_NAME` - app1/0'
```
## Documentation changes
None

## Bug reference
https://bugs.launchpad.net/juju/+bug/1708492
